### ISSUE TITLE
Cannot download source code of jpcap

### ISSUE THREAD

**Issue by huiyiqun** [WEZARD/pangu_ansible#2]
SPEC文件里面给的下载网站是这个：`http://netresearch.ics.uci.edu/`，但是这个网站完全打不开，没有这个网站。

据我所知，jpcap的项目是host在[这个页面上](http://sourceforge.net/projects/jpcap/)的，为什么不直接从这里下载代码呢？

另外从changelog上看到的版本号是0.7，然而实际上jpcap并没有这个版本，你准备下载的是哪个版本的呢？

从jpcap的SPEC文件看，你是直接把源代码解开之后就打包起来了，但是这个是源代码而不是发布文件，你这样把源代码直接放到目标机器上肯定是不能直接使用的，这个和storm这样的纯java项目是有很大的区别的，这点需要注意一下。

**Comment by WEZARD** [comment]
我提供那个网站出现在好多人的博客中，我的 jpcap 0.7 是参考

```
http://www.sipinspector.com/download/jpcap
```

然后我也看了一下您提供的网站，获取了一下 .src.rpm 的 spec文件，发现它的打包过程也是只进行了编译的操作，而我只需要 libjpcap.so 这一个文件（其余的我们都做了外部依赖），我感觉我把它放到 javalibrary 的路径下就可以了，之前刘同学也测试过，这么做可以正常工作，并且提供的 jpcap 版本是不是太久远了。

**Comment by huiyiqun** [comment]
所以你直接找了别人编译的lib文件？这个不可接受

**Comment by huiyiqun** [comment]
如果是官方发布的二进制文件也罢了 第三方发布的二进制文件不一定安全

**Comment by WEZARD** [comment]
不是。。。我只用了官方发布的 libjpcap.so 这一个文件，然后放到 javalibrary 路径下，这样打包的时候source 可以不用填写源码网址，将这个文件填写在 source 里即可。

**Comment by WEZARD** [comment]
并且这个文件很小，附加进去也可以

**Comment by huiyiqun** [comment]
官方并没有给出x86_64的二进制发布 你给的链接并不是jpcap的官方链接

**Comment by huiyiqun** [comment]
x86_64

**Comment by huiyiqun** [comment]
这不是文件大小的问题 而是来源安全性的问题 如果_官方_提供了二进制的发布 那么使用官方的二进制发布是可以的 否则当然是用_官方_的源代码进行编译和打包

> 官方的spec只进行了编译

源代码里不可能有libjpcap.so 不编译当然没有这个文件

**Comment by WEZARD** [comment]
![qq20150827-2 2x](https://cloud.githubusercontent.com/assets/13159665/9513243/0d320b44-4cc0-11e5-961e-81d9b4547d76.jpg)

**Comment by huiyiqun** [comment]
你看一下官方的spec里是怎么编译的吧 jpcap这个项目我也没有接触过

**Comment by WEZARD** [comment]
我这个就是完全按照官方的步骤来的。。
那个网站的讨论区也有一样的问题，但是没有解答。

**Comment by WEZARD** [comment]
我发现 jpcap 文件的 README 中指明了 libjpcap.so
![qq20150828-1 2x](https://cloud.githubusercontent.com/assets/13159665/9537663/a6a4fa36-4d6d-11e5-86e0-1f086a4c0927.jpg)
这样的话是不是可以单独把这个文件打包进去了，现在官方给的编译报错。

**Comment by WEZARD** [comment]
详细内容：
![qq20150828-2 2x](https://cloud.githubusercontent.com/assets/13159665/9537699/1899d51c-4d6e-11e5-929e-a40069e3e8d0.jpg)

**Comment by huiyiqun** [comment]
那个so文件是32位的，最好编译一个64位的。

**Comment by huiyiqun** [comment]
关于如何编译在`path/to/jpcap/docs/BUILD.unix`，里面讲到了需要`source`一个bash scripts，之后才能编译。

**Comment by huiyiqun** [comment]
我看了一下 [wikipedia](https://en.wikipedia.org/wiki/Pcap)，我给你的链接所指向的项目叫[jpcap](http://jpcap.sourceforge.net/)，而你之前试图打包的那个项目叫[Jpcap](http://netresearch.ics.uci.edu/kfujii/Jpcap/doc/index.html)，看起来都挺久没有更新了，当然也可能是很稳定了。

看起来你需要的是`Jpcap`而不是`jpcap`。

关于`Jpcap`，这个项目最后一次更新是2007年，上上次是2005，且项目主页已经没有了，让我觉得很悲伤。你下载源代码的地方是[SIP Inspector](http://www.sipinspector.com/)，我并没有找到任何证据证明这个项目和`Jpcap`有任何关联，它之所以host `Jpcap`的源代码貌似仅仅是因为它依赖`Jpcap`并且`Jpcap`的原始主页死掉了。

所以如果你要继续给这个死掉的项目打包，可能只能把SPEC里的Source写成[SIP Inspector](http://www.sipinspector.com/)的地址了。

其实`Pcap`的java wrapper其实还有几个，[其中有一个](https://github.com/kaitoy/pcap4j)看上去稍微活跃一些，不过我不太清楚你们的工程依赖`Jpcap`的原因，所以可能这个并不适合。

**Comment by WEZARD** [comment]
这个网站提供了 rpm 包（第一个）

```
http://www.sipinspector.com/download/jpcap
```

安装的时候提示：
![qq20150828-4 2x](https://cloud.githubusercontent.com/assets/13159665/9541630/b4e43cb8-4d9c-11e5-87cf-aeddf797cb59.jpg)
但是已经使用高于 1.6 的版本了
![qq20150828-5 2x](https://cloud.githubusercontent.com/assets/13159665/9541646/d36c3a14-4d9c-11e5-95fe-257965ebe18e.jpg)

**Comment by huiyiqun** [comment]
应该是因为打包的时间比较久远了 依赖的包的名字变了
